Utility/PiecesTemplates.FCMacro

```diff
@@ -0,0 +1,421 @@
+        "Informazioni per l'uso:",
+        "Gruppi/Cartelle da creare:",
+        "Crea la struttura della Parte",
+        "Versione:",
+        "Descrizione:",
+        "Informazioni",
+        "Questo script (macro) genera una struttura di gruppi (cartelle) in una nuova parte, in modo che sia più semplice organizzare le varie operazioni, i riferimenti e gli elementi di questa parte.",
+        "I gruppi inizialmente risultano predefiniti, però si possono modificare facilmente.",
+        "Operazioni sui Gruppi",
+        "Aggiungi Gruppo",
+        "Elimina Gruppo",
+        "per primo",
+        "per ultimo",
+        "Lista delle parti create:",
+        "Nº di parti create:" 
+    ]
+        ]
+
+ui_path = app.getUserMacroDir(True) + "/PiecesTemplates/ui/PiecesTemplates.ui"
+gr_path = app.getUserMacroDir(True) + "/PiecesTemplates/Groups.txt"
+
+try:
+    _fromUtf8 = QtCore.QString.fromUtf8
+except AttributeError:
+    def _fromUtf8(s):
+        return s
+
+class PiecesTemplates:
+
+    def __init__(self):
+        self.doc = app.activeDocument()
+        if not self.doc:
+            app.Console.PrintWarning(tr('El documento no está activo') + cIntro)
+            return
+        
+        # Init var
+        self.form = gui.PySideUic.loadUi(ui_path)        
+        self.numeroPiezas = 0
+        self.pieza = ""
+        self.form.botonCrear.setEnabled(False)
+
+        # Open File "Groups.txt"
+        self.openF()
+        
+        # Init UI with content
+        self.mostrarCheckbox()
+        self.initStrings()
+
+        # Connect
+        self.form.botonCrear.clicked.connect(self.createPart)        
+        self.form.edicionNombrePieza.returnPressed.connect(self.createPart)
+        self.form.edicionNombrePieza.textChanged.connect(self.nombrePiezaCambia)
+        self.form.botonAyuda.clicked.connect(self.mostrarHelp)        
+        self.form.eliminarGrupo.clicked.connect(self.on_eliminarGrupo_clicked)
+        self.form.anadirGrupo.clicked.connect(self.on_anadirGrupo_clicked)
+
+    def openF(self):
+        with open(gr_path, "r") as groupsFile:
+            for linea in groupsFile.readlines():
+                linea = re.sub("\n","",linea)
+                if (linea != ""):
+                    listGroups.append(linea)
+                    self.form.comboEliminarGrupo.addItem(linea)
+        if (len(listGroups) > maxElem):
+            etError1 = __Name__ + "(Error)"
+            error1 = "En el archivo de texto hay más líneas de las permitidas." + " "
+            error1 += "El máximo es: " + str(maxElem) + cIntro + cIntro
+            error1 += "Se aconseja cancelar y corregir el archivo de texto:" + cIntro + cIntro
+            error1 += gr_path + cIntro + cIntro
+            reply = QtGui.QMessageBox.warning(None,etError1,error1)
+        self.longitudListaGrupos = len(listGroups)
+
+    def translate(self, text):
+        if (text not in strLang[0]) or (g_num_lang < 0) or (g_num_lang >= len(strLang)):
```

Comment:
You can combine as `not (0 < g_num_lang <= len(strLang))`, which I find clearer.